sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -499,7 +500,18 @@ private boolean usesContext(AdditionalParameter param) {
      */
     private Constructor<? extends DoFnInvoker<?, ?>> createInvokerConstructor(
         @SuppressWarnings("rawtypes") Class<? extends DoFnWithContext> clazz) {
+
+      final TypeDescription clazzDescription = new TypeDescription.ForLoadedType(clazz);
+
       DynamicType.Builder<?> builder = new ByteBuddy()
+          // Create subclasses inside the target class's package
+          .with(new NamingStrategy.SuffixingRandom("auxiliary") {
+                @Override
+                public String subclass(TypeDescription.Generic superClass) {
+                  return super.name(clazzDescription);
```

Comment:
I think that is accurate, but is the default behavior for ByteBuddy more than my choice. I think it shows up as `org.apache.beam.blah.bah.DoFn$$ByteBuddy$$lkasdjfsladf`.

Very open to other naming possibilities, but I'm inclined to always delegate to bytebuddy's underlying naming strategies, and they handle some corner cases (that maybe don't really apply to us, but I only browsed the code briefly). In the event that bytebuddy fixes issues in their own naming strategies, I'd like to inherit the fixes, too.